test/integration/apiserver/apply/reset_fields_test.go

```diff
@@ -119,8 +119,10 @@ var resetFieldsSpecData = map[schema.GroupVersionResource]string{
 	gvr("certificates.k8s.io", "v1beta1", "certificatesigningrequests"):            `{}`,
 	gvr("flowcontrol.apiserver.k8s.io", "v1alpha1", "flowschemas"):                 `{"metadata": {"labels":{"a":"c"}}, "spec": {"priorityLevelConfiguration": {"name": "name2"}}}`,
 	gvr("flowcontrol.apiserver.k8s.io", "v1beta1", "flowschemas"):                  `{"metadata": {"labels":{"a":"c"}}, "spec": {"priorityLevelConfiguration": {"name": "name2"}}}`,
+	gvr("flowcontrol.apiserver.k8s.io", "v1beta2", "flowschemas"):                  `{"metadata": {"labels":{"a":"c"}}, "spec": {"priorityLevelConfiguration": {"name": "name2"}}}`,
 	gvr("flowcontrol.apiserver.k8s.io", "v1alpha1", "prioritylevelconfigurations"): `{"metadata": {"labels":{"a":"c"}}, "spec": {"limited": {"assuredConcurrencyShares": 23}}}`,
 	gvr("flowcontrol.apiserver.k8s.io", "v1beta1", "prioritylevelconfigurations"):  `{"metadata": {"labels":{"a":"c"}}, "spec": {"limited": {"assuredConcurrencyShares": 23}}}`,
+	gvr("flowcontrol.apiserver.k8s.io", "v1beta2", "prioritylevelconfigurations"):  `{"metadata": {"labels":{"a":"c"}}, "spec": {"limited": {"assuredConcurrencyShares": 23}}}`,
```

Comment:
This PR introduces `v1beta2` for priority & fairness - `v1beta2` is a straight copy from `v1beta1` - no changes in the API spec. 

`TestApplyResetFields` (integration test) is failing for `v1beta2` of `priorityLevelConfiguration` with the following error
```
reset_fields_test.go:265: Failed to apply obj2: Apply failed with 4 conflicts: conflicts with "fieldmanager1":
        - .spec.limited.assuredConcurrencyShares
        - .status.conditions[type="MyStatus"]
        - .status.conditions[type="MyStatus"].type
        - .status.conditions[type="MyStatus"].type
```
but the same test for `v1beta1` is assing.

@apelisse  I am not familiar with the inner working of server side apply, i would appreciate it if you can provide any pointer for me to troubleshoot this issue further.

Reply:
Very glad to see the integration test caught the gap. @apelisse, it would be good to document what people should do here and point them to the docs from the test failure

Looks like v1beta2 cases are needed in these methods:
* flowSchemaStrategy#GetResetFields()
* flowSchemaStatusStrategy#GetResetFields()
* priorityLevelConfigurationStrategy#GetResetFields()
* priorityLevelConfigurationStatusStrategy#GetResetFields()

Reply:
thanks @liggitt, I added the  required `GetResetFields` fields for `v1beta2` and now the test is passing.

Reply:
also, opened a new issue https://github.com/kubernetes/kubernetes/issues/104842 [  `TestApplyResetFields` test skips `flowschemas` resource due to flake]. 

But I ran it locally to make sure reset fields for `flowschemas` `v1beta2` work as expected.